packages/account/src/providers/provider.ts

```diff
@@ -680,45 +691,42 @@ export default class Provider {
    * @param transactionRequest - The transaction request object.
    * @returns A promise.
    */
-  async estimateTxDependencies(transactionRequest: TransactionRequest): Promise<void> {
-    let missingOutputVariableCount = 0;
-    let missingOutputContractIdsCount = 0;
-    let tries = 0;
-
+  async estimateTxDependencies(transactionRequest: TransactionRequest): Promise<CallResult> {
     if (transactionRequest.type === TransactionType.Create) {
-      return;
+      return {
+        receipts: [],
+      };
     }
 
-    let txRequest = transactionRequest;
+    await this.estimatePredicates(transactionRequest);
 
-    if (txRequest.hasPredicateInput()) {
-      txRequest = (await this.estimatePredicates(txRequest)) as ScriptTransactionRequest;
-    }
+    let receipts: TransactionResultReceipt[] = [];
 
-    while (tries < MAX_RETRIES) {
+    for (let attempt = 0; attempt < MAX_RETRIES; attempt++) {
       const { dryRun: gqlReceipts } = await this.operations.dryRun({
-        encodedTransaction: hexlify(txRequest.toTransactionBytes()),
+        encodedTransaction: hexlify(transactionRequest.toTransactionBytes()),
         utxoValidation: false,
       });
-      const receipts = gqlReceipts.map(processGqlReceipt);
+      receipts = gqlReceipts.map(processGqlReceipt);
```

Comment:
The `receipts` variable is always assigned the last dry run's receipts. If the last dry run was successful, the receipts are those of a successful one and all is good. If the last dry run is unsuccessful and no additional dry run was made due to the `MAX_RETRIES` constraint, the receipts will be those of an unsuccessful one and that will be caught higher up the line.